docs/usage/sc_AIRRseq/single_cell_tutorial.md

```diff
@@ -0,0 +1,214 @@
+
+The analysis steps and their corresponding folders, where the results are stored, are briefly listed below. Detailed documentation on the pipeline output can be found on the [Output documentation page](https://nf-co.re/airrflow/docs/output/).
+
+1. QC and sequence assembly (if starting from fastq files).
+   - In this first step, Cell Ranger's VDJ algorithm is employed to assemble contigs, annotate contigs, call cells and generate clonoytpes. The results are stored in the 'cellranger' folder.
+
+2. V(D)J annotation and filtering.
+   - In this step, gene segments are assigned using a germline reference. Alignments are annotated in AIRR format. Non-productive sequences and sequences with low alignment quality are removed. Metadata is added. The results are under the folder named 'vdj_annotation'.
+
+3. QC filtering.
+   - In this step, cells without heavy chains or with multiple heavy chains are removed. Sequences in different samples that share the same cell_id and necleotide sequence are filtered out. The result are stored in the 'qc-filtering' folder.
+
+4. Clonal analysis.
+   - In this step, the Hamming distance threshold of the junction regions is determined when clonal_threshold is set to 'auto' (by default). It should be reviewed for accuracy once the result is out. The threshold result can be found under the folder clonal_analysis/find_threshold.
+   - If the automatic threshold is unsatisfactory, you can set the threshold manually and re-run the pipeline.
+   (Tip: use -resume whenever running the Nextflow pipeline to avoid duplicating previous work).
+   - For TCR data, where somatic hypermutation does not occur, set the clonal_threshold to 0 when running the Airrflow pipeline.
+   - Once the threshold is established, clones are assigned to the sequences. A variety of tables and plots associated with clonal analysis were added to the folder `clonal_analysis/define_clones`, such as  sequences_per_locus_table, sequences_per_c_call_table, sequences_per_constant_region_table, num_clones_table, clone_sizes_table,clone size distribution plot, clonal abundance plot, diversity plot and etc.
+
+5. Repertoire analysis.
+   - Calculation of several repertoire characteristics, e.g. V gene usage, for comparison between subjects, time points or cell populations. The output folder is `repertoire_comparison`.
+
+6. Other reporting.
+   Additional reports are also generated, including:
+   - MultiQC report: summarizes QC metrics across all samples.
+   - Pipeline_info report: various reports relevant to the running and execution of the pipeline.
+   - Report_file_size report: Summary of the number of sequences left after each of the most important pipeline steps.
+
+## Understanding error messages
```

Comment:
@Vivian0105 I just thought of a new section to explain how to understand error messages and facilitate debugging. What do you think?